lib/full-site-editing/template-loader.php

```diff
@@ -291,3 +291,16 @@ function gutenberg_get_edit_template_link( $link, $post_id ) {
 	return admin_url( sprintf( $edit_link, urlencode( $template->id ), $template->type ) );
 }
 add_filter( 'get_edit_post_link', 'gutenberg_get_edit_template_link', 10, 2 );
+
+/**
+ * Remove the emoji script for themes that support block templates.
+ *
+ * @return void
+ */
+function _gutenberg_maybe_remove_emoji_detection_script() {
+	if ( gutenberg_supports_block_templates() ) {
```

Comment:
I wonder if we should check this or its subset (is_block_theme)

Reply:
WP doesn't support IE or older browsers anymore, so the assumption is that if a theme supports block templates, it's up to date - which is a given if they use the latest Gutenberg plugin version.
Ideally the script would be removed on **all** sites, not just block-based ones - and I'm about to submit a patch in core to do just that. The emojis script has been a painful thorn for performance for a looong time, now that it's no longer needed it can be removed. Sites that need it and explicitly have to support 10-year-old browsers can manually add the script. Though that is highly unlikely to ever happen... Sites that need to stay compatible with older browsers are traditionally poorly maintained, banking or otherwise official sites where emojis will never be used 👍